openedx/features/edly/validators.py

```diff
@@ -42,3 +46,25 @@ def is_edly_user_allowed_to_login(request, possibly_authenticated_user):
         return True
 
     return False
+
+
+def is_edly_user_allowed_to_login_with_social_auth(request, user):
+    """
+    Check if the user is allowed to login on the current site with social auth.
+
+    Arguments:
+        request (object): HTTP request object
+        user: User object trying to authenticate
+
+    Returns:
+        bool: Returns True if User can login to site otherwise False.
+    """
+
+    if not is_edly_user_allowed_to_login(request, user):
```

Comment:
@burhandodhy can we remove this logic from here and from `login_user` method and move it into `set_logged_in_edly_cookies` method

Reply:
@ziafazal  I did move code from `login_user` to `set_logged_in_edly_cookies` but it create an issue. As `set_logged_in_edly_cookies` runs after user is authenticated successfully https://github.com/edly-io/edx-platform/blob/37a3b5167bbfb472018dd118cc531dff5d4e7ba5/openedx/core/djangoapps/user_authn/views/login.py#L370 

So In a case where does user doesn't have specific site access, LMS show the error message as expected but when you click the login button second, `EdlyOrganizationAccessMiddleware` middleware refresh the page as it found that user is already loggedIn.